conan/internal/runner/docker.py

```diff
@@ -64,13 +64,28 @@ def __init__(self, conan_api, command, host_profile, build_profile, args, raw_ar
         import docker
         import docker.api.build
         try:
-            self.docker_client = docker.from_env()
-            self.docker_api = docker.APIClient()
+            try:
+                self.docker_client = docker.from_env()
+                self.docker_api = docker.APIClient()
+            except:
+                docker_base_urls = [
+                    os.environ.get('CONAN_RUNNER_DOCKER_HOST'),
+                    os.environ.get('DOCKER_HOST'),
+                    'unix:///var/run/docker.sock',
+                    f'unix://{os.path.expanduser("~")}/.rd/docker.sock'
+                ]
+                for base_url in docker_base_urls:
```

Comment:
Sounds a bit hacky, some suggestions:
- Add some comment explaining the why. Explain the different URLs a bit
- Is it possible to unify the above ``docker.from_env()`` and this call in a single one? so trying to connect the docker client looks like less like a workaround hack?